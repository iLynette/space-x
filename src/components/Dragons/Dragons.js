const Dragons = () => <div />;
export default Dragons;
