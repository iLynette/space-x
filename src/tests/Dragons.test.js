import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Dragons, { Dragon } from '../components/Dragons/Dragons';

describe('Dragon tests', () => {
  it('render test', () => {
    const dragon = {
      id: 1,
      image: 'https://img.icons8.com/external-xnimrodx-blue-xnimrodx/64/000000/external-rocket-online-marketing-xnimrodx-blue-xnimrodx.png',
      name: 'dragon1',
      type: 'capsule',
      description: 'description for dragon1',
      reserved: false,
    };

    const tree = renderer.create(
      <Provider store={store}>
        <Dragon
          id={dragon.id}
          image={dragon.image}
          name={dragon.name}
          type={dragon.type}
          description={dragon.description}
          reserved={dragon.reserved}
        />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Test the whole functionality of dragons page', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Dragons />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

it('Test the whole functionality of dragons page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Dragons />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
