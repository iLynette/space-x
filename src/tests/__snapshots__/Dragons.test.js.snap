// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon tests Test the whole functionality of dragons page 1`] = `null`;

exports[`Dragon tests render test 1`] = `
<div
  className="dragonCard"
>
  <img
    alt="dragon"
    className="dragonImage"
    src="https://img.icons8.com/external-xnimrodx-blue-xnimrodx/64/000000/external-rocket-online-marketing-xnimrodx-blue-xnimrodx.png"
  />
  <div
    className="dragonInfo"
  >
    <h2>
      dragon1
    </h2>
    <h3
      className="dragonType"
    >
      capsule
    </h3>
    <p>
      description for dragon1
    </p>
    <button
      className="reservDragon"
      id="1"
      onClick={[Function]}
      type="button"
    >
      Reserve
    </button>
  </div>
</div>
`;
